python/desc/slrealizer/plotting.py

```diff
@@ -37,57 +34,92 @@ def determineScale(lensX, sourceX, lensY, sourceY):
     plt.title('Observation with ' + 'filter ' + currObs[1] + ' on MJD ' + str(currObs[0]))
+
+    scale_factor = 2
+    PSF_HWHM = currObs[2]/scale_factor
+
+    # Draw quasar images:
     for i in range(4):
-        source = plt.Circle((sourceX, sourceY), radius=currObs[2]/scale_factor, alpha=quasar_alpha, fc=circleColor, linewidth=0)
-        if(convolve):
-            # For now, we just let lensFWHM to be zero.
-            #lensFWHM = currLens['REFF']
-            lensFWHM = 0.0
-            lens = plt.Circle((lensX[i]+sourceX, lensY[i]+sourceY), radius=convolve(currObs[2], lensFWHM)/scale_factor, alpha=lens_alpha, fc=circleColor, linewidth=0)
-        else:
-            lens = plt.Circle((lensX[i]+sourceX, lensY[i]+sourceY), radius=currObs[2]/scale_factor, alpha=lens_alpha, fc=circleColor, linewidth=0, edgecolor='b', linestyle='dashed')
-        plt.gca().add_patch(lens)
-        plt.gca().add_patch(source)
+        source = plt.Circle((sourceX, sourceY),
+                            radius=PSF_HWHM,
+                            alpha=quasar_alpha,
+                            fc=circleColor, linewidth=0)
+    # Draw lens galaxy:
+    if(convolve):
+        # For now, we just let lensFWHM to be zero.
+        #lensFWHM = currLens['REFF']
+        lensFWHM = 0.0
+        galaxyHWHM = convolve(currObs[2], lensFWHM)/scale_factor
+        # Bug: the LENS position is in the CENTER of the field.
+        #      The SOURCE position is UNOBSERVABLE.
+        #      The IMAGE positions are in teh OM10 catalog in the
+        #      XIMG and YIMG columns.
+        lens = plt.Circle((lensX[i]+sourceX, lensY[i]+sourceY),
+                          radius=galaxy_HWHM,
+                          alpha=lens_alpha,
+                          fc=circleColor, linewidth=0)
+    else:
+        lens = plt.Circle((lensX[i]+sourceX, lensY[i]+sourceY),
+                          radius=PSF_HWHM,
+                          alpha=lens_alpha,
+                          fc=circleColor, linewidth=0,
+                          edgecolor='b', linestyle='dashed')
+    plt.gca().add_patch(lens)
+    plt.gca().add_patch(source)
 
-        #seeing = plt.Circle(((-plotY-2*currObs[2])/scale_factor, (plotX-2*currObs[2])/scale_factor), radius=currObs[2]/scale_factor, alpha=0.1, fc='k')
-    seeing = plt.Circle((-2.5, -2.5), radius=currObs[2]/scale_factor, alpha=0.1, fc='k')
-    plt.legend((source, seeing, lens),('source', 'seeing', 'lens'))
+        #seeing = plt.Circle(((-plotY-2*currObs[2])/scale_factor, (plotX-2*currObs[2])/scale_factor), radius=PSF_HWHM, alpha=0.1, fc='k')
+    seeing = plt.Circle((-2.5, -2.5),
+                        radius=PSF_HWHM,
+                        alpha=0.1,
+                        fc='black')
+    plt.legend((source, seeing, lens),
+               ('QSO images', 'PSF', 'Lens galaxy'), fontsize=10)
     plt.gca().add_patch(seeing)
 
 
-def drawPlot(currObs, currLens, convolve = False):
+def draw_model(currObs, currLens, convolve=False):
```

Comment:
I'm not sure if the to-be-written methods `draw_sources`, draw_diasources` etc can re-use this plotting code - they might be able to, in which case you would rename the method to just `draw` and then provide a kwarg to choose what type of features it draws (model components, Sources, DIASources etc)